File: backend/app/utils/game_aliases.py
"""Game name aliases: profile/onboarding use full names (e.g. Assetto Corsa Competizione), admin events use short (e.g. ACC)."""

GAME_ALIASES_LONG_TO_SHORT = {
    "Assetto Corsa Competizione": "ACC",
    "Assetto Corsa": "AC",
}

def expand_driver_games_for_event_match(driver_games: list) -> list:
    """Include short and long names so Event.game matches driver sim_games (either format)."""
    out = list(driver_games) if driver_games else []
    for long_name, short_name in GAME_ALIASES_LONG_TO_SHORT.items():
        if long_name in out and short_name not in out:
            out.append(short_name)
        if short_name in out and long_name not in out:
            out.append(long_name)
    return out

File: frontend/src/utils/gameAliases.js
/**
 * Event.game can be short (ACC, AC); driver.sim_games use full names (Assetto Corsa Competizione, Assetto Corsa).
 * Treat these as the same when matching.
 */
const GAME_ALIASES = {
  ACC: ['Assetto Corsa Competizione'],
  'Assetto Corsa Competizione': ['ACC'],
  AC: ['Assetto Corsa'],
  'Assetto Corsa': ['AC'],
};

/**
 * Normalize game string for comparison (lowercase, trim).
 */
function normalizeGame(g) {
  return (g || '').trim().toLowerCase();
}

/**
 * All names that should match the given game (the name itself + aliases).
 */
export function namesForGame(game) {
  const g = (game || '').trim();
  if (!g) return [];
  const lower = g.toLowerCase();
  const out = new Set([g, lower]);
  const aliases = GAME_ALIASES[g] || GAME_ALIASES[g.toLowerCase()];
  if (aliases) aliases.forEach((a) => out.add(a) && out.add(a.toLowerCase()));
  return [...out];
}

/**
 * Returns true if event's game matches any of driver's sim_games (considering aliases).
 */
export function eventGameMatchesDriverGames(eventGame, driverSimGames) {
  if (!eventGame || !driverSimGames?.length) return false;
  const driverSet = new Set((driverSimGames || []).map(normalizeGame));
  const possible = namesForGame(eventGame);
  return possible.some((name) => driverSet.has(normalizeGame(name)));
}
